server-common/src/main/java/org/apache/kafka/server/util/LockUtils.java

```diff
@@ -49,4 +57,73 @@ public static <T> T inLock(Lock lock, Supplier<T> supplier) {
             lock.unlock();
         }
     }
+
+    /**
+     * Executes the given {@link Runnable} within the context of the specified {@link Lock}.
+     * The lock is acquired before executing the runnable and released after the execution,
+     * ensuring that the lock is always released, even if an exception is thrown.
+     *
+     * @param lock     the lock to be acquired and released
+     * @param runnable the runnable to be executed within the lock context
+     * @throws NullPointerException if either {@code lock} or {@code runnable} is null
+     */
+    public static void inLock(Lock lock, Runnable runnable) {
```

Comment:
Excuse me, have we considered replacing `inLock` by `inLockThrows`? Keeping both varieties seems a bit verbose to me.